packages/runtime/test-runtime-utils/src/mocks.ts

```diff
@@ -124,47 +169,124 @@ export class MockContainerRuntime {
 		// Set FluidDataStoreRuntime's deltaManager to ours so that they are in sync.
 		this.dataStoreRuntime.deltaManager = this.deltaManager;
 		this.dataStoreRuntime.quorum = factory.quorum;
+		this.dataStoreRuntime.containerRuntime = this;
 		// FluidDataStoreRuntime already creates a clientId, reuse that so they are in sync.
 		this.clientId = this.dataStoreRuntime.clientId ?? uuid();
 		factory.quorum.addMember(this.clientId, {});
+		this.runtimeOptions = makeContainerRuntimeOptions(mockContainerRuntimeOptions);
+		assert(
+			this.runtimeOptions.flushMode !== FlushMode.Immediate ||
+				!this.runtimeOptions.enableGroupedBatching,
+			"Grouped batching is not compatible with FlushMode.Immediate",
+		);
 	}
 
 	public createDeltaConnection(): MockDeltaConnection {
-		const deltaConnection = new MockDeltaConnection(
-			(messageContent: any, localOpMetadata: unknown) =>
-				this.submit(messageContent, localOpMetadata),
-			() => this.dirty(),
+		assert(
+			this.dataStoreRuntime.createDeltaConnection !== undefined,
+			"Unsupported datastore runtime version",
 		);
+		const deltaConnection = this.dataStoreRuntime.createDeltaConnection();
 		this.deltaConnections.push(deltaConnection);
 		return deltaConnection;
 	}
 
 	public submit(messageContent: any, localOpMetadata: unknown): number {
-		const clientSequenceNumber = this.clientSequenceNumber++;
-		const msg: Partial<ISequencedDocumentMessage> = {
-			clientId: this.clientId,
-			clientSequenceNumber,
-			contents: messageContent,
-			referenceSequenceNumber: this.deltaManager.lastSequenceNumber,
-			type: MessageType.Operation,
-		};
-		this.factory.pushMessage(msg);
+		const clientSequenceNumber = this.clientSequenceNumber;
+		switch (this.runtimeOptions.flushMode) {
+			case FlushMode.Immediate: {
+				this.factory.pushMessage({
```

Comment:
can we get code reuse with `submitInternal` here?

Reply:
great catch!